satpy/tests/reader_tests/test_ascat_l2_soilmoisture_bufr.py

```diff
@@ -0,0 +1,180 @@
+        'hour': 9,
+        'minute': 33,
+        'second': np.linspace(0, 59, samples),
+        'latitude': lat,
+        'longitude': lon,
+        'surfaceSoilMoisture': surfaceSoilMoisture,
+        'soilMoistureQuality': np.zeros(samples),
+    }
+    return retmsg
+
+
+msg = create_message()
+
+# the notional filename that would contain the above test message data
+FILENAME = 'W_XX-EUMETSAT-TEST,SOUNDING+SATELLITE,METOPA+ASCAT_C_EUMC_20201221093300_73545_eps_o_125_ssm_l2.bin'
+# the information that would be extracted from the above filename according to the pattern in the .yaml
+FILENAME_INFO = {
+    'reception_location': 'TEST',
+    'platform': 'METOPA',
+    'instrument': 'ASCAT',
+    'start_time': '20201221093300',
+    'perigee': '73545',
+    'species': '125_ssm',
+    'level': 'l2'
+}
+
+# file type info for the above file that is defined in the .yaml
+FILETYPE_INFO = {
+    'file_type': 'ascat_l2_soilmoisture_bufr',
+    'file_reader': 'ASCATSOILMOISTUREBUFR'
+}
+
+
+def save_test_data(path):
+    """Save the test file to the indicated directory."""
+    import eccodes as ec
+    filepath = os.path.join(path, FILENAME)
+    with open(filepath, "wb") as f:
+        for m in [msg]:
+            buf = ec.codes_bufr_new_from_samples('BUFR4_local_satellite')
+            for key in m:
+                val = m[key]
+                if np.isscalar(val):
+                    ec.codes_set(buf, key, val)
+                else:
+                    ec.codes_set_array(buf, key, val)
+            ec.codes_set(buf, 'pack', 1)
+            ec.codes_write(buf, f)
+            ec.codes_release(buf)
+    return filepath
+
+
+class TesitAscatL2SoilmoistureBufr(unittest.TestCase):
+    """Test ASCAT Soil Mosture loader."""
+
+    def setUp(self):
+        """Create temporary file to perform tests with."""
+        import tempfile
+        from satpy.readers.ascat_l2_soilmoisture_bufr import ASCATSOILMOISTUREBUFR
+        self.base_dir = tempfile.mkdtemp()
+        self.fname = save_test_data(self.base_dir)
+        self.fname_info = FILENAME_INFO
+        self.ftype_info = FILETYPE_INFO
+        self.reader = ASCATSOILMOISTUREBUFR(self.fname, self.fname_info, self.ftype_info)
+
+    def tearDown(self):
+        """Remove the temporary directory created for a test."""
+        try:
+            import shutil
+            shutil.rmtree(self.base_dir, ignore_errors=True)
+        except OSError:
+            pass
+
+    @unittest.skipIf(sys.platform.startswith('win'), "'eccodes' not supported on Windows")
+    def test_scene(self):
+        """Test scene creation."""
+        from satpy import Scene
+        fname = os.path.join(self.base_dir, FILENAME)
+        scn = Scene(reader='ascat_l2_soilmoisture_bufr', filenames=[fname])
+        self.assertTrue('start_time' in scn.attrs)
+        self.assertTrue('end_time' in scn.attrs)
+        self.assertTrue('sensor' in scn.attrs)
+        self.assertTrue('scatterometer' in scn.attrs['sensor'])
```

Comment:
Would be nice to check the actual values of the attributes to make sure they are interpreted correctly

Reply:
I added test for the values that are read from the file: start_time & end_time